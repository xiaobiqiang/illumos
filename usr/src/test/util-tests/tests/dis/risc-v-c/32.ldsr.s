/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test RV32C-specific loads and stores.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	c.flw	fs1,(a0)
	c.flw	fa3,0x40(a3)
	c.flwsp	fs1,(sp)
	c.flwsp	ft8,0x40(sp)
	c.flwsp	fa5,0x34(sp)
	c.fswsp	ft2,(sp)
	c.fswsp	fa3,0x20(sp)
.size libdis_test, [.-libdis_test]
