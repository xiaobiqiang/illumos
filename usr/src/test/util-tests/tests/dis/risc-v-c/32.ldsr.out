    libdis_test:     04 61              c.flw fs1,0x0(a0)
    libdis_test+0x2: b4 62              c.flw fa3,0x40(a3)
    libdis_test+0x4: 82 64              c.flwsp fs1,0x0(sp)
    libdis_test+0x6: 06 6e              c.flwsp ft8,0x40(sp)
    libdis_test+0x8: d2 77              c.flwsp fa5,0x34(sp)
    libdis_test+0xa: 0a e0              c.fswsp ft2,0x0(sp)
    libdis_test+0xc: 36 f0              c.fswsp fa3,0x20(sp)
